### PR TITLE
Consider custom theme when painting vertical tabs

### PR DESCRIPTION
**Pull request by sangwoo108**
Resolves https://github.com/brave/brave-browser/issues/28335

image

image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Manual
1. Go to brave://settings/manageProfile and pick a theme color
2. Vertical tabs should be visible regardless of OS theme settings(light, dark)

### CODE REVIEW

File: browser/ui/color/brave_color_mixer.cc [hunk 1]
```diff
@@ -279,10 +280,7 @@ void AddBraveLightThemeColorMixer(ui::ColorProvider* provider,
   mixer[kColorFeaturePromoBubbleCloseButtonInkDrop] = {
       GetToolbarInkDropColor(mixer)};
 
-  mixer[kColorBraveVerticalTabSeparator] = {SkColorSetRGB(0xE2, 0xE3, 0xE7)};
-  mixer[kColorBraveVerticalTabActiveBackground] = {SK_ColorWHITE};
-  mixer[kColorBraveVerticalTabInactiveBackground] = {
-      SkColorSetRGB(0xf3, 0xf3, 0xf3)};
+  tabs::AddBraveVerticalTabLightThemeColorMixer(provider, key);
```
Comment (simonhong) [inline browser/ui/color/brave_color_mixer.cc:283]: Maybe this should be guarded with in `base::FeatureList::IsEnabled(tabs::features::kBraveVerticalTabs)`?

File: browser/ui/color/brave_color_mixer.cc [hunk 2]
```diff
@@ -376,11 +374,7 @@ void AddBraveDarkThemeColorMixer(ui::ColorProvider* provider,
   mixer[kColorFeaturePromoBubbleCloseButtonInkDrop] = {
       GetToolbarInkDropColor(mixer)};
 
-  mixer[kColorBraveVerticalTabSeparator] = {SkColorSetRGB(0x5E, 0x61, 0x75)};
-  mixer[kColorBraveVerticalTabActiveBackground] = {
-      SkColorSetRGB(0x18, 0x1A, 0x21)};
-  mixer[kColorBraveVerticalTabInactiveBackground] = {
-      SkColorSetRGB(0x30, 0x34, 0x43)};
+  tabs::AddBraveVerticalTabDarkThemeColorMixer(provider, key);
```
Comment (simonhong) [inline browser/ui/color/brave_color_mixer.cc:377]: ditto

**Review by simonhong — APPROVED** [review]
++

File: browser/ui/color/brave_color_mixer.cc (hunk 1 above)
Comment (sangwoo108) [reply browser/ui/color/brave_color_mixer.cc:283]: Good call!

File: browser/ui/tabs/features.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_TABS_FEATURES_H_
+#define BRAVE_BROWSER_UI_TABS_FEATURES_H_
+
+#include "base/feature_list.h"
+
+namespace tabs::features {
```
Comment (simonhong) [inline browser/ui/tabs/features.h:11]: How about making one features file here by moving another feature(`kBraveChangeActiveTabOnScrollEvent`) here?
and then `browser/ui/views/tabs/features.cc` could be renamed such as `browser/ui/views/tabs/utils.cc`?

Comment (sangwoo108) [reply browser/ui/tabs/features.h:11]: In the previous review, I got feedback that we should keep a feature in views/ if the feature is used only by the views/. So I feel like keeping kBraveChangeActiveTabOnScrollEvent there for now.

Comment (simonhong) [reply browser/ui/tabs/features.h:11]: Well.. it's weird to me to have two feature files for one feature(`tabs`).
If we move it also, I think we can avoid view path including(brave/browser/ui/views/tabs/features.h) at `brave/browser/about_flags.cc` and can use `brave/browser/ui/tabs/features` instead.
I think it's more cleaner?

Comment (sangwoo108) [reply browser/ui/tabs/features.h:11]: Yeah, that's what I prefer, too. Let's get this done then.

File: chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc [hunk 4]
```diff
@@ -39,7 +40,8 @@
 #undef GetBrowserViewForNativeWindow
 #undef TabDragController
 
-#include "brave/browser/ui/views/tabs/features.h"
+#include "brave/browser/ui/tabs/features.h"
+#include "brave/browser/ui/views/tabs/vertical_tab_utils.h"
```
Comment (mkarolin) [inline chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc:44]: Do we need to include these again?

**Review by mkarolin — APPROVED** [review]
`chromium_src` changes LGTM (with a question).

File: chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc (hunk 4 above)
Comment (sangwoo108) [reply chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc:44]: Oops, will remove it.

**Pull request merged**